scenarioo-client/app/shared/navigation/selectedBranchAndBuild.service.js

```diff
@@ -99,13 +99,25 @@ angular.module('scenarioo.services').factory('SelectedBranchAndBuildService', fu
     }
 
     function registerSelectionChangeCallback(callback) {
-        selectionChangeCallbacks.push(callback);
+        addCallback(selectionChangeCallbacks, callback);
         var selected = getSelectedBranchAndBuild();
         if (isBranchAndBuildDefined()) {
             callback(selected);
         }
     }
 
+    function addCallback(callbackList, newCallback) {
+        var callbackListContainsNewCallback = false;
+        angular.forEach(callbackList, function(callback){
+            if(callback.toString() === newCallback.toString()) {
+                callbackListContainsNewCallback = true;
+            }
+        });
+        if(!callbackListContainsNewCallback) {
+            callbackList.push(newCallback);
+        }
+    }
```

Comment:
Gut gesehen. Der Code scheint einen Callback nur hinzuzufügen, wenn er noch nicht in der Liste ist. Wahrscheinlich soll damit verhindert werden, dass man zu viele Events kriegt, bei einem Selection Change. Der Service ist ein Singleton und behält seinen State. Die Controllers werden neu instanziert bei jedem Seitenwechsel. Dies ist nur eine Vermutung.

Das ganze Konzept lösen wir wohl besser durch events, statt Callbacks ab. Was meinst du @bruderol ?